internal/collection/collection.go

```diff
@@ -90,8 +90,13 @@ func (svc *collectionImpl) Create(ctx context.Context, col *Collection) error {
 }
 
 func (svc *collectionImpl) CheckDuplicate(ctx context.Context, name string) (bool, error) {
-	// raw sql here.
-	return false, nil
+	query := "SELECT EXISTS(SELECT 1 FROM collection WHERE name = ? AND completed_at IS NOT NULL)"
```

Comment:
Should we do something like the following instead?

    SELECT EXISTS(SELECT 1 FROM collection WHERE name = ? AND status NOT IN (3, 6))

So when looking for a duplicate we would ignore 3 (StatusError) and 6 (StatusAbandoned).

Reply:
Yes I was going to ask about this. I like it